test/foundry/deploy/10_holograph_tests.t.sol

```diff
@@ -0,0 +1,700 @@
+    address user = vm.addr(2);
+    uint256 privateKeyDeployer = 0xff22437ccbedfffafa93a9f1da2e8c19c1711052799acf3b58ae5bebb5c6bd7b;
+    address deployer = vm.addr(privateKeyDeployer);
+
+    bytes initCode;
+    uint32  holographChainId;
+    address bridge;
+    address factory;
+    address interfaces;
+    address operator;
+    address registry;
+    address treasury;
+    address utilityToken;
+
+    Holograph holograph;
+    MockExternalCall mockExternalCall;
+
+    function randomAddress() public view returns (address) {
+        uint256 randomNum = uint256(keccak256(abi.encodePacked(block.timestamp, block.difficulty)));
+        return address(uint160(randomNum));
+    }
+
+    function setUp() public {
+    
+    // Deploy contracts
+    vm.startPrank(deployer);
+    holograph = new Holograph();
+    mockExternalCall = new MockExternalCall();
+    vm.stopPrank();
+
+    holographChainId = 1;
+    bridge = randomAddress();
+    factory = randomAddress();
+    interfaces = randomAddress();
+    operator = randomAddress();
+    registry = randomAddress();
+    treasury = randomAddress();
+    utilityToken = randomAddress();
+
+    bytes memory initCode = generateInitCode(holographChainId, bridge, factory, interfaces, operator, registry, treasury, utilityToken);
+    holograph.init(initCode);
+    }
+
+    function generateInitCode(
+    uint32 holographChainId,
+    address bridge,
+    address factory,
+    address interfaces,
+    address operator,
+    address registry,
+    address treasury,
+    address utilityToken
+    ) public pure returns (bytes memory) {
+        return abi.encode(
+        holographChainId, bridge, factory, interfaces, operator, registry, treasury, utilityToken
+        );
+    }
+
+/*
+    INIT()
+*/
+
+    /**
+    * @notice Test the initialization of the Holograph contract.
+    * @dev This is a basic test to ensure the initialization process works as expected.
+    * This test deploys a new instance of the Holograph contract, generates initialization code,
+    * and initializes the contract with the provided parameters.
+    */
+    function testInit() public {
+        Holograph holographTest;
+        holographTest = new Holograph();
+        bytes memory initCode = generateInitCode(holographChainId, bridge, factory, interfaces, operator, registry, treasury, utilityToken);
+        holographTest.init(initCode);
+    }
+
+    /**
+    * @notice Test the setAdmin function of the Holograph contract.
+    * @dev This test is designed to verify the functionality of setting a new admin address in the contract.
+    * This test retrieves the current admin address from the Holograph contract,
+    * performs a prank operation on the admin address using the VM,
+    * and then sets a new admin address to the contract.
+    */
+    // TODO Admin does not correspond to the address that performs the init
+    function testSetAdmin() public {
+        admin = holograph.getAdmin();
+        vm.prank(admin);
```

Comment:
use the new var origin in all these cases where it is necessary to change the contract call to the owner.